src/fsharp/Interactive.DependencyManager/DependencyManager.fs

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Interactive.DependencyManager
+
+open System
+open System.IO
+open System.Reflection
+open FSharp.Reflection
+open Internal.Utilities.FSharpEnvironment
+
+module ReflectionHelper =
+    let dependencyManagerPattern = "*DependencyManager*.dll"
+    let dependencyManagerAttributeName= "DependencyManagerAttribute"
+    let resolveDependenciesMethodName = "ResolveDependencies"
+    let namePropertyName = "Name"
+    let keyPropertyName = "Key"
+    let seqEmpty = Seq.empty<string>
+
+    let assemblyHasAttribute (theAssembly: Assembly) attributeName =
+        try
+            CustomAttributeExtensions.GetCustomAttributes(theAssembly)
+            |> Seq.exists (fun a -> a.GetType().Name = attributeName)
+        with | _ -> false
+
+    let getAttributeNamed (theType: Type) attributeName =
+        try
+            theType.GetTypeInfo().GetCustomAttributes false
+            |> Seq.tryFind (fun a -> a.GetType().Name = attributeName)
+        with | _ -> None
+
+    let getInstanceProperty<'treturn> (theType: Type) propertyName =
+        try
+            let property = theType.GetProperty(propertyName, typeof<'treturn>)
+            if isNull property then
+                None
+            elif not (property.GetGetMethod().IsStatic)
+                 && property.GetIndexParameters() = Array.empty
+            then
+                Some property
+            else
+                None
+        with | _ -> None
+
+    let getInstanceMethod<'treturn> (theType: Type) (parameterTypes: Type array) methodName =
+        try
+            let theMethod = theType.GetMethod(methodName, parameterTypes)
+            if isNull theMethod then
+                None
+            else
+                Some theMethod
+        with | _ -> None
+
+    let stripTieWrapper (e:Exception) =
+        match e with
+        | :? TargetInvocationException as e->
+            e.InnerException
+        | _ -> e
+
+open ReflectionHelper
+
+(* Shape of Dependency Manager contract, resolved using reflection *)
+[<AllowNullLiteralAttribute>]
+type IDependencyManagerProvider =
+    abstract Name: string
+    abstract Key: string
+    abstract ResolveDependencies: scriptDir: string * mainScriptName: string * scriptName: string * scriptExt: string * packageManagerTextLines: string seq * tfm: string -> bool * string seq * string seq * string seq
+
+[<RequireQualifiedAccess>]
+type ErrorReportType =
+| Warning
+| Error
```

Comment:
Nit: four spaces

Reply:
oooh ... whilst I don't often have formatting peeves, indenting enum cases is one of them.  Every time I see it, it makes me grind my teeth.  But in the spirit of comity I will make the change.